### DISCUSSION TITLE
Overriding the configuration of a Quarkus app from its test code

### DISCUSSION THREAD

**Discussion by giscus** [quarkusio/quarkus#41661] [Quarkus Blog/Website]
# Overriding the configuration of a Quarkus app from its test code

Quarkus: Supersonic Subatomic Java

https://quarkus.io/blog/overriding-configuration-from-test-code/?ref=dailydev

**Comment by billapepper** [comment]
Thanks for this guide! I have been stuck due to a Quarkus bug https://github.com/quarkusio/quarkus/issues/38774 with @TestProfiles and metaspace usage... but I should be able to rewrite the tests using Mockito outlined in this blog to get around it.

**Reply by gwenneg** [reply]
You're welcome! 😄

**Comment by DeMol-EE** [comment]
You could consider another approach based on MicroProfile: registering a custom ConfigSource. With a little effort, you can make sure it is reset before each test (using a quarkus before each hook SPI) and you can hide the logic to look up the custom source (needed to override a specific value in a test) behind a static method. It's similar to the Mockito approach, except without Mockito.

**Comment by fkellner** [comment]
I tried the approach described in section [Quarkus Test Profiles](https://quarkus.io/blog/overriding-configuration-from-test-code/#approach-1-quarkus-test-profiles), under
> The test class which is also a test profile

But it results in a `java.lang.NoSuchMethodException: [...].FeaturesResourceTest.<init>()`, both when run from IntelliJ with the Quarkus Extension and run from Maven. What could I be doing wrong?

**Reply by fkellner** [reply]
Adding a public constructor only results in 
```
java.lang.IllegalAccessException: class io.quarkus.test.junit.AbstractJvmQuarkusTestExtension cannot access a member of class [...].FeaturesResourceTest with modifiers "public"
```

**Reply by fkellner** [reply]
Would have been neat, but putting this method in a separate class to be used as Profile is not too much work either.

**Reply by gwenneg** [reply]
Here's an implementation of that approach that still works with the latest Quarkus release: [FeaturesResourceTestProfileTest.java](https://github.com/gwenneg/blog-overriding-configuration-from-test-code/blob/main/test-profile/src/test/java/com/gwenneg/blog/config/FeaturesResourceTestProfileTest.java).

**Comment by mizalewski** [comment]
Wow! Thank you for explaining possible solutions for configuration. I try to implement DSL with drivers from Dave Farley approach for acceptance tests and I learned a lot from this article! ♥️
